rplugin/python3/deoplete/sources/buffer.py

```diff
@@ -39,22 +39,25 @@ def on_buffer(self, context):
 
     def __make_cache(self, context, bufnr):
         p = re.compile(context['keyword_patterns'])
+        buffer = self.vim.current.buffer
+        for b in reversed(self.vim.buffers):
+            if b.number < bufnr:
+                buffer = b
+                break
 
-        bufnr -= 1
```

Comment:
I think I misunderstood this.  Is the cache being created for the current buffer?